ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ConfigManagementServiceIT.scala

```diff
@@ -177,4 +186,89 @@ final class ConfigManagementServiceIT extends LedgerTestSuite {
     }
   })
 
+  test(
+    "DuplicateSubmissionIdWhenSubmissionsCorrect",
+    "Duplicate submission ids are accepted when config changed twice",
+    allocate(NoParties, NoParties),
+    runConcurrently = false,
+  )(implicit ec => { case Participants(Participant(alpha), Participant(beta)) =>
+    // Multiple config changes should never cause duplicates. Participant adds extra entropy to the
+    // submission id to ensure client does not inadvertently cause problems by poor selection
+    // of submission ids.
+    for {
+      (req1, req2) <- generateRequest(alpha).map(r =>
+        r -> r
+          .update(_.configurationGeneration := r.configurationGeneration + 1)
+      )
+      _ <- ignoreNotAuthorized(alpha.setTimeModel(req1))
+      _ <- ignoreNotAuthorized(beta.setTimeModel(req2))
```

Comment:
Does this work on distributed ledgers?

The first line only returns after the config has been persisted on the ledger (because the first participant waits for the config change event).
However, it might take some time until index of the second participant has caught up with the new config. If it hasn't caught up, the check in `ApiConfigManagementService.setTimeModel` will fail.

The situation is complicated because we verify the config generation both in the participant server (`ApiConfigManagementService`, which relies on information from the index that is not up-to-date) and on the ledger (which should rely on the correct information).

Reply:
Very good point! I will synchronize in between

Reply:
Done